src/components/settings/DiabeloopHistoryParameters.js

```diff
@@ -118,73 +118,88 @@ export default class HistoryTable extends Table {
     );
   }
 
-  formatData(data, params, ord) {
-    const history = data.parameters;
-    let currentParameters = params;
-    let order = ord;
-    const formattedData = _.map(history, (hist) => {
-      const h = hist;
-      switch (h.changeType) {
-        case 'added':
-          currentParameters.push(h);
-          break;
-        case 'deleted':
-          currentParameters = _.filter(currentParameters, s => s.name !== h.name);
-          break;
-        case 'updated':
-          currentParameters = _.map(currentParameters, current => {
-            const param = current;
-            if (param.name === h.name) {
-              h.previousValue = param.value;
-              h.previousUnit = param.unit;
-              param.value = h.value;
-              param.unit = h.unit;
+  getAllRows(history) {
+    const rows = [];
+
+    if (_.isArray(history)) {
+      const currentParameters = new Map();
+
+      const nHistory = history.length;
+      for (let i = 0; i < nHistory; i++) {
+        const parameters = history[i].parameters;
+
+        if (_.isArray(parameters)) {
+          const nParameters = parameters.length;
+          let latestDate = new Date(0);
+
+          // Compare b->a since there is a reverse order at the end
+          parameters.sort((a, b) => b.name.localeCompare(a.name));
```

Comment:
As we have now different levels of parameters, I just realized we should sort on 2 fields: level and name. Can you apply the below change:
```suggestion
          parameters.sort((a, b) =>  
            b.level.toString().localeCompare(a.level.toString())
            || b.name.localeCompare(a.name)
            );
```